category/functor.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2017 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Simon Hudon
+
+Standard identity and composition functors
+-/
+import tactic.ext
+import category.basic
+
+universe variables u v w u' v' w'
+
+section functor
+
+variables {F : Type u → Type v}
+variables {α β γ : Type u}
+variables [functor F] [is_lawful_functor F]
+
+lemma functor.map_id : functor.map id = (id : F α → F α) :=
```

Comment:
Is there a reason why you proof something about `functor.map` and not `<$>`?

Reply:
The reason is if you write `functor.map id <$> x` you can reduce it to `x` using `functor.map_id` and `is_lawful_functor.id_map` (or `functor.map_id` twice plus unfolding `id`)